### DISCUSSION TITLE
Send telemetry trace via. rabbitmq and start a new scope with specific telemetry trace

### DISCUSSION THREAD

**Discussion by SalomonHenke** [microsoft/aspire#1414] [Q&A] [answered]
Hey all. Limited knowledge of opentelemetry. 

Is there a way to get a scopes telemetry trace? And then, is there a way to start a new scope via. the IServiceProvider and provide it with the telemetry id? Trying to keep a trace accros services. Idea was to send it via. the rabbitmq message / header param. But still need to be able to pick up the trace again + get the trace in the first place.

 Images for reference. Havent found any obvious solutions online. 
![image](https://github.com/dotnet/aspire/assets/96744651/ba0075de-bb06-4afd-bf49-a8a75aaa8f62)

**Comment by DamianEdwards** [comment]
You do this in code using `ActivitySource`. Take a look at the [documentation here](https://learn.microsoft.com/dotnet/core/diagnostics/distributed-tracing-instrumentation-walkthroughs).

**Reply by davidfowl** [reply]
An example of doing this with rabbitmq is here https://github.com/dotnet/eShop/blob/3b2515e69c16b1c2f64276aef3caeca5f947d399/src/EventBusRabbitMQ/RabbitMQEventBus.cs

**Comment by DGAISmith** [comment]
Is this something which could be added to the RabbitMq Aspire Component at some point in the future, or is not how the team sees the Component responsibilities developing over time?

**Reply by davidfowl** [reply, marked as answer]
I think it depends on how this progresses https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1261
